### PR TITLE
test: Command palette unit tests crd handler

### LINKED ISSUE #5064: Add unit tests for CommandPalette navigation handlers
**Description**
The command palette handlers in src/command-pallette/CommandPalletteUI/handlers/ contain ~666 statements of pure navigation and search logic with 0% unit test coverage. This is a high-risk area as it handles resource navigation, fuzzy search, and keyboard interactions entirely in logic functions that are straightforward to unit-test in isolation.

**Reasons**
The command palette is a core power-user feature used to navigate across all Kubernetes resource types. Its handlers are pure functions with no DOM dependency, making them ideal unit test targets. Without tests, regressions in search ranking, resource filtering, or navigation path construction can go undetected until integration testing or user reports. Adding unit tests here directly improves confidence when refactoring the handlers or extending the palette with new resource types.

**Acceptance Criteria**
- [ ] Unit tests cover the main handler functions (navigation, resource lookup, search filtering)
- [ ] Each handler function has at least one passing unit test covering the happy path
- [ ] Edge cases are covered: empty input, unknown commands, permission-restricted commands
- [ ] Statement coverage for `src/command-pallette/CommandPalletteUI/handlers` is higher than the current one
- [ ] Branch coverage is higher than the current one

**Testing Strategy (DoD)**

- Automated unit tests (Vitest) for all handler functions in src/command-pallette/CommandPalletteUI/handlers/

**Attachments**
Current coverage (from report dated 2026-06-16):
- Statements: 0/666
- Branches: 0/358
- Functions: 0/223

**Related Issues**
#5071

### PR DESCRIPTION
**Pull request by dbadura**
**Description**

Changes proposed in this pull request:

- add unit tests for crdHandler
- add unit tests for one helper function

**Related issue(s)**
https://github.com/kyma-project/busola/issues/5064

**Definition of done**

- [x] The PR's title starts with one of the following prefixes:
  - feat: A new feature
  - fix: A bug fix
  - docs: Documentation only changes
  - refactor: A code change that neither fixes a bug nor adds a feature
  - test: Adding tests
  - revert: Revert commit
  - chore: Maintainance changes to the build process or auxiliary tools, libraries, workflows, etc.
- [x] Related issues are linked. To link internal trackers, use the issue IDs like `backlog#4567`
- [x] Explain clearly why you created the PR and what changes it introduces
- [x] All necessary steps are delivered, for example, tests, documentation, merging

### CODE REVIEW

File: src/command-pallette/CommandPalletteUI/handlers/test/helpers.test.ts [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+      tokens: ['cred'],
+      resources: [],
+      resourceTypeNames: ['crd'],
+    };
+    const result = getSuggestionForSingleResource(ctx);
+
+    expect(result).toBe('crd');
+  });
+
+  it('Random tokens, no suggetions', () => {
```
Comment (OliwiaGowor) [inline src/command-pallette/CommandPalletteUI/handlers/test/helpers.test.ts]:
```suggestion
  it('Random tokens, no suggestions', () => {
```

File: src/command-pallette/CommandPalletteUI/handlers/test/crdHandler.test.ts [hunk 2]
```diff
@@ -0,0 +1,212 @@
...
+  });
+
+  it('AutoComplete given 2 tokens returns empty result', () => {
+    const ctx = fixContext({
+      tokens: ['crd', 'xyz'],
+      resourceCache: {
+        customresourcedefinitions: [
+          fixCRD('abcd'),
+          fixCRD('xyzResource'),
+          fixCRD('resoruceXyz'),
```
Comment (OliwiaGowor) [inline src/command-pallette/CommandPalletteUI/handlers/test/crdHandler.test.ts]:
```suggestion
          fixCRD('resourceXyz'),
```

File: src/command-pallette/CommandPalletteUI/handlers/test/helpers.test.ts [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+    const ctx = {
+      tokens: ['random-toekns'],
```
Comment (OliwiaGowor) [inline src/command-pallette/CommandPalletteUI/handlers/test/helpers.test.ts]:
```suggestion
      tokens: ['random-tokens'],
```

**Review by OliwiaGowor — COMMENTED** [review]
-[ ] All tests pre-populate `resourceCache.customresourcedefinitions`, so the case where the cache hasn't loaded yet is not tested
- [x] In CRDs tests there is no check for when unrelated token is provided, e.g. pods - `createResults` should return empty in this case

**Review by OliwiaGowor — APPROVED** [review]
